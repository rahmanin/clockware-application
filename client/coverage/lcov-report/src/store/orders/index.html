
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/store/orders</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../index.html">All files</a> src/store/orders</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">88.64% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>39/44</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">76.92% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>10/13</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">76.92% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>10/13</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">90.24% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>37/41</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line high'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file medium" data-value="actions.ts"><a href="actions.ts.html">actions.ts</a></td>
	<td data-value="73.68" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 73%"></div><div class="cover-empty" style="width: 27%"></div></div>
	</td>
	<td data-value="73.68" class="pct medium">73.68%</td>
	<td data-value="19" class="abs medium">14/19</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="70" class="pct medium">70%</td>
	<td data-value="10" class="abs medium">7/10</td>
	<td data-value="77.78" class="pct medium">77.78%</td>
	<td data-value="18" class="abs medium">14/18</td>
	</tr>

<tr>
	<td class="file high" data-value="reducer.ts"><a href="reducer.ts.html">reducer.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="25" class="abs high">25/25</td>
	<td data-value="81.82" class="pct high">81.82%</td>
	<td data-value="11" class="abs high">9/11</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="23" class="abs high">23/23</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Tue Dec 01 2020 12:22:24 GMT+0200 (Eastern European Standard Time)
            </div>
        </div>
        <script src="../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../sorter.js"></script>
        <script src="../../../block-navigation.js"></script>
    </body>
</html>
    