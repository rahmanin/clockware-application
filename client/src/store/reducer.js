import { combineReducers } from "redux";
import pricesReducer from "./prices/reducer";

const rootReducer = combineReducers({
  prices: pricesReducer,
});

export default rootReducer;