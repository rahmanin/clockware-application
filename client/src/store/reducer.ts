import { combineReducers } from "redux";
import pricesReducer from "./prices/reducer";
import usersReducer from "./users/reducer";
import citiesReducer from "./cities/reducer";
import mastersReducer from "./masters/reducer";
import ordersReducer from "./orders/reducer";
import ordersClientReducer from "./ordersClient/reducer";
import { createStore, applyMiddleware } from "redux";
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension'
import newsReducer from "./news/reducer";

const rootReducer = combineReducers({
  prices: pricesReducer,
  user: usersReducer,
  cities: citiesReducer,
  masters: mastersReducer,
  orders: ordersReducer,
  ordersClient: ordersClientReducer,
  news: newsReducer
});

const composedEnhancer = composeWithDevTools(
  applyMiddleware(thunk)
)

const store = createStore(
  rootReducer,
  composedEnhancer
);

export default store
