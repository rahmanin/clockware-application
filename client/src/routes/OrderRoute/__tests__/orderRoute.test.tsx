import React from 'react';
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import {cleanup} from '@testing-library/react';
import { Provider } from "react-redux";
import store from "../../../store/reducer";
import { act } from "react-dom/test-utils";
import {mount} from "enzyme"
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';
import MakingOrder from '../index';

Enzyme.configure({ adapter: new Adapter() });
afterEach(cleanup);

const middlewares = [thunk]
const mockStore = configureMockStore(middlewares)

act(() => {
  it('Render Loader when loading = true', () => {
    const component = mount(
      <Provider store={store}>
        <MakingOrder/>
      </Provider>
    );
    expect(component.debug()).toMatchSnapshot();
  });

  it("Should toggle loading on false and render whole component", () => {
    const newStore = mockStore(
      {
        cities: { 
          list: [{id:3, city: "aaaaa"}], 
          loading: false 
        },
        user: { 
          list: [{id:1, role: "admin"}], 
          loading: false 
        },
        prices: { 
          list: [{id:1, size: "any", price: 666}], 
          loading: false 
        },
      })
    const component = mount(
      <Provider store={newStore}>
        <MakingOrder/>
      </Provider>
    );
    expect(component.debug()).toMatchSnapshot();
  });
})
