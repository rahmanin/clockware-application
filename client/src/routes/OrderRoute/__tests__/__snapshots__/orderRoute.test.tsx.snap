// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Loader when loading = true 1`] = `
"<Provider store={{...}}>
  <MakingOrder>
    <Loader>
      <div className=\\"isLoading\\">
        <div className=\\"circle c_1\\" />
        <div className=\\"circle c_2\\" />
        <div className=\\"circle c_3\\" />
      </div>
    </Loader>
  </MakingOrder>
</Provider>"
`;

exports[`Should toggle loading on false and render whole component 1`] = `
"<Provider store={{...}}>
  <MakingOrder>
    <div className=\\"order_wrapper\\">
      <h1>
        To make an order, fill the form:
      </h1>
      <form className=\\"orderForm\\">
        <label htmlFor=\\"username\\">
          Name
        </label>
        <input className=\\"field\\" required={true} id=\\"username\\" name=\\"username\\" type=\\"text\\" disabled={false} onChange={[Function (anonymous)]} onBlur={[Function (anonymous)]} value=\\"\\" />
        <label htmlFor=\\"email\\">
          Email Address
        </label>
        <input required={true} className=\\"field\\" id=\\"email\\" disabled={false} name=\\"email\\" type=\\"email\\" onChange={[Function (anonymous)]} onBlur={[Function (anonymous)]} value=\\"\\" />
        <label htmlFor=\\"city\\">
          City
        </label>
        <select required={true} className=\\"field\\" id=\\"city\\" name=\\"city\\" onChange={[Function (anonymous)]} value=\\"aaaaa\\">
          <option>
            aaaaa
          </option>
        </select>
        <label htmlFor=\\"sizePrice\\">
          Size, price (hrn)
        </label>
        <select required={true} className=\\"field\\" id=\\"sizePrice\\" name=\\"sizePrice\\" onChange={[Function (anonymous)]} value=\\"any, 666\\">
          <option>
            any, 666
          </option>
        </select>
        <label htmlFor=\\"order_date\\">
          Date
        </label>
        <input required={true} className=\\"field\\" id=\\"order_date\\" name=\\"order_date\\" type=\\"date\\" onChange={[Function (anonymous)]} onBlur={[Function (anonymous)]} min=\\"2020-12-14\\" value=\\"2020-12-14\\" />
        <label htmlFor=\\"order_time_start\\">
          Time
        </label>
        <select required={true} className=\\"field\\" id=\\"order_time_start\\" name=\\"order_time_start\\" onBlur={[Function (anonymous)]} onChange={[Function (anonymous)]} value=\\"14:00\\">
          <option hidden={true} disabled={true}>
            8
            :00
          </option>
          <option hidden={true} disabled={true}>
            9
            :00
          </option>
          <option hidden={true} disabled={true}>
            10
            :00
          </option>
          <option hidden={true} disabled={true}>
            11
            :00
          </option>
          <option hidden={true} disabled={true}>
            12
            :00
          </option>
          <option hidden={true} disabled={true}>
            13
            :00
          </option>
          <option hidden={false} disabled={false}>
            14
            :00
          </option>
          <option hidden={false} disabled={false}>
            15
            :00
          </option>
          <option hidden={false} disabled={false}>
            16
            :00
          </option>
          <option hidden={false} disabled={false}>
            17
            :00
          </option>
        </select>
        <label>
          Add image (1mb, jpg or png)
        </label>
        <Upload className=\\"choose_img_btn\\" beforeUpload={[Function: beforeUpload]} accept=\\".jpg, .png\\" onChange={[Function: onChange]} openFileDialogOnClick={true} type=\\"select\\" multiple={false} action=\\"\\" data={{...}} showUploadList={true} listType=\\"text\\" disabled={false} supportServerRender={true}>
          <span className=\\"choose_img_btn\\">
            <div className=\\"ant-upload ant-upload-select ant-upload-select-text\\" style={[undefined]}>
              <Upload onStart={[Function: onStart]} onError={[Function: onError]} onProgress={[Function: onProgress]} onSuccess={[Function: onSuccess]} beforeUpload={[Function: beforeUpload]} accept=\\".jpg, .png\\" onChange={[Function: onChange]} openFileDialogOnClick={true} type=\\"select\\" multiple={false} action=\\"\\" data={{...}} showUploadList={true} listType=\\"text\\" disabled={false} supportServerRender={true} prefixCls=\\"ant-upload\\" component=\\"span\\" headers={{...}} name=\\"file\\" multipart={false} customRequest={{...}} withCredentials={false}>
                <AjaxUploader onStart={[Function: onStart]} onError={[Function: onError]} onProgress={[Function: onProgress]} onSuccess={[Function: onSuccess]} beforeUpload={[Function: beforeUpload]} accept=\\".jpg, .png\\" onChange={[Function: onChange]} openFileDialogOnClick={true} type=\\"select\\" multiple={false} action=\\"\\" data={{...}} showUploadList={true} listType=\\"text\\" disabled={false} supportServerRender={true} prefixCls=\\"ant-upload\\" component=\\"span\\" headers={{...}} name=\\"file\\" multipart={false} customRequest={{...}} withCredentials={false}>
                  <span onClick={[Function (anonymous)]} onKeyDown={[Function (anonymous)]} onMouseEnter={[undefined]} onMouseLeave={[undefined]} onDrop={[Function (anonymous)]} onDragOver={[Function (anonymous)]} tabIndex=\\"0\\" className=\\"ant-upload\\" role=\\"button\\" style={[undefined]}>
                    <input id={[undefined]} type=\\"file\\" onClick={[Function: onClick]} style={{...}} accept=\\".jpg, .png\\" multiple={false} onChange={[Function (anonymous)]} />
                    <Button loading={false} ghost={false} block={false} htmlType=\\"button\\">
                      <Wave>
                        <button type=\\"button\\" className=\\"ant-btn\\" onClick={[Function: handleClick]}>
                          <LoadingIcon existIcon={false} prefixCls=\\"ant-btn\\" loading={false}>
                            <CSSMotion visible={false} motionName=\\"ant-btn-loading-icon-motion\\" removeOnLeave={true} onAppearStart={[Function: getCollapsedWidth]} onAppearActive={[Function: getRealWidth]} onEnterStart={[Function: getCollapsedWidth]} onEnterActive={[Function: getRealWidth]} onLeaveStart={[Function: getRealWidth]} onLeaveActive={[Function: getCollapsedWidth]}>
                              <DomWrapper />
                            </CSSMotion>
                          </LoadingIcon>
                          <span>
                            Choose image
                          </span>
                        </button>
                      </Wave>
                    </Button>
                  </span>
                </AjaxUploader>
              </Upload>
            </div>
            <LocaleReceiver componentName=\\"Upload\\" defaultLocale={{...}}>
              <UploadList listType=\\"text\\" items={{...}} previewFile={[Function: previewImage]} onPreview={[undefined]} onDownload={[undefined]} onRemove={[Function: handleRemove]} showRemoveIcon={true} showPreviewIcon={true} showDownloadIcon={false} removeIcon={[undefined]} downloadIcon={[undefined]} iconRender={[undefined]} locale={{...}} isImageUrl={[Function: isImageUrl]} progress={{...}} appendAction={[undefined]} itemRender={[undefined]}>
                <div className=\\"ant-upload-list ant-upload-list-text\\">
                  <CSSMotionList motionName=\\"ant-upload-animate\\" onAppearStart={[Function: getCollapsedHeight]} onEnterStart={[Function: getCollapsedHeight]} onAppearActive={[Function: getRealHeight]} onEnterActive={[Function: getRealHeight]} onLeaveStart={[Function: getCurrentHeight]} onLeaveActive={[Function: getCollapsedHeight]} motionDeadline={500} keys={{...}} motionAppear={true} component={false} />
                </div>
              </UploadList>
            </LocaleReceiver>
          </span>
        </Upload>
        <Button type=\\"button\\" color=\\"black\\" title=\\"Find your master\\" onClick={[Function: formSubmit]} disabled={true}>
          <button className=\\"button button--black\\" disabled={true} type=\\"button\\" onClick={[Function: formSubmit]}>
            Find your master
          </button>
        </Button>
      </form>
    </div>
  </MakingOrder>
</Provider>"
`;
